core/services/offchainreporting/oracle.go

```diff
@@ -210,19 +214,35 @@ type dataSource struct {
 
 var _ ocrtypes.DataSource = (*dataSource)(nil)
 
+// The context passed in here has a timeout of observationTimeout.
+// Gorm/pgx doesn't return a helpful error upon cancellation, so we manually check for cancellation and return a
+// appropriate error.
 func (ds dataSource) Observe(ctx context.Context) (ocrtypes.Observation, error) {
+	start := time.Now()
 	runID, err := ds.pipelineRunner.CreateRun(ctx, ds.jobID, nil)
+	endCreate := time.Now()
+	if ctx.Err() != nil {
```

Comment:
Is this thread-safe?

Reply:
ya see https://github.com/golang/go/blob/master/src/context/context.go#L370